beps/0001-notifications-system/README.md

```diff
@@ -160,89 +168,202 @@ The following backend service interfaces are added as part of this proposal.
 ```ts
 // TODO - We may want to add an additional wrapping here with interfaces for Notification and NotificationParameters
 
-interface SendNotificationRequest {
-  receivers: SignalReceivers;
+export type NotificationRecipients =
+  | {
+      type: 'entity';
+      entityRef: string | string[]; // typically a user or group entity, otherwise `spec.owner` of the entity/entities
+    }
+  | {
+      type: 'broadcast'; // all users
+    };
 
-  /* TODO - please contribute :) */
-}
+export type NotificationType = 'undone' | 'done' | 'saved';
 
-interface SendNotificationResponse {
+export type Notification = {
   id: string;
-  receivers: SignalReceivers;
+  userRef: string;
+  title: string;
+  description: string;
+  origin: string;
+  link: string;
+  created: Date;
+  saved: boolean;
+  read?: Date;
+  topic?: string;
+  updated?: Date;
+  icon?: string;
+  image?: string;
```

Comment:
What do we need both of these for?

Reply:
For Web Notification API

Reply:
Can you describe a particular use case?

Reply:
One of the things mentioned in this BEP is the usage of https://developer.mozilla.org/en-US/docs/Web/API/Notification which can be enriched with an image. It doesn't know anything about icons, so we should keep the option there. Also, there is a reference implementation for using this API in the initial PR.

Reply:
Hmm, it's not impossible that we'd be able to serialize icons as a data URI. I think it could be good to hold off on images for now since it introduces quite a bit of complexity. Should be fairly straightforward to add support for that later I'm thinking?